tests/test_benchmark_simple_scenarios.py

```diff
@@ -125,46 +135,40 @@ def test_benchmark_ABE_grid_pv_bat(self, margs):
         # compare the total excess electricity between the two cases
         assert excess["AB_grid_PV"] < excess["ABE_grid_PV_battery"]
 
-        @pytest.mark.skipif(
-            EXECUTE_TESTS_ON not in (TESTS_ON_MASTER),
-            reason="Benchmark test deactivated, set env variable "
-            "EXECUTE_TESTS_ON to 'master' to run this test",
+    @pytest.mark.skipif(
+        EXECUTE_TESTS_ON not in (TESTS_ON_MASTER),
+        reason="Benchmark test deactivated, set env variable "
+        "EXECUTE_TESTS_ON to 'master' to run this test",
+    )
+    @mock.patch("argparse.ArgumentParser.parse_args", return_value=argparse.Namespace())
+    def test_benchmark_AD_grid_diesel(self, margs):
+        # define the two cases needed for comparison (grid + PV) and (grid + PV + battery)
+        use_case = "AD_grid_diesel"
+        # define an empty dictionary for excess electricity
+        main(
+            overwrite=True,
+            display_output="warning",
+            path_input_folder=os.path.join(TEST_INPUT_PATH, use_case),
+            input_type=CSV_EXT,
+            path_output_folder=os.path.join(TEST_OUTPUT_PATH, use_case),
         )
-        @mock.patch(
-            "argparse.ArgumentParser.parse_args", return_value=argparse.Namespace()
+
+        # read json with results file
+        with open(
+            os.path.join(TEST_OUTPUT_PATH, use_case, "json_with_results.json"), "r"
+        ) as results:
+            data = json.load(results)
```

Comment:
why don't you use our parser (`B0.load_json`) ?

Reply:
I copied it from somewhere above, I tought it was the right one. Will fix it.